src/lib.rs

```diff
@@ -102,22 +97,48 @@ mod app;
 pub use app::App;
 
 mod errors {
+    use ffi_utils::StringError;
+    use std::io;
     use std::str::Utf8Error;
 
-    error_chain! {
-        types {
-            Error, ErrorKind, ChainErr, Result;
-        }
-
-        foreign_links {
-            Utf8Error(Utf8Error);
-        }
-
-        errors {
-            /// The SystemURI error used to wrap problems
-            SystemUriError(t: String) {
-                description("System URI Error")
-                display("Could not execute: {}", t)
+    quick_error! {
+        /// System URI error variants.
+        #[derive(Debug)]
+        pub enum Error {
+            /// IO error.
+            IoError(error: io::Error) {
+                description("Io error")
```

Comment:
The `description` method is soft-deprecated and we have a task to remove all instances across all our crates.